### PR TITLE
Linkable tree

### PR DESCRIPTION
**Pull request by lazovicff**

### CODE REVIEW

File: pallets/anchor/src/lib.rs [hunk 1]
```diff
@@ -416,158 +313,24 @@ impl<T: Config<I>, I: 'static> AnchorInterface<AnchorConfigration<T, I>> for Pal
...
 		}
 		let result = <T as pallet::Config<I>>::Verifier::verify(&bytes, proof_bytes)?;
 		ensure!(result, Error::<T, I>::InvalidWithdrawProof);
 		// transfer the assets
+		let anchor = Self::get_anchor(id)?;
 		<T as Config<I>>::Currency::transfer(anchor.asset, &Self::account_id(), &recipient, anchor.deposit_size)?;
 		Ok(())
 	}
 
-	fn add_edge(
```
Comment (drewstone) [inline pallets/anchor/src/lib.rs:429]: Can we keep these functions mocked through the `T::LinkableTree` interface?

File: pallets/anchor/src/lib.rs [hunk 2]
```diff
@@ -416,158 +313,24 @@ impl<T: Config<I>, I: 'static> AnchorInterface<AnchorConfigration<T, I>> for Pal
...
-			(id, src_chain_id),
-			neighbor_root_inx + T::RootIndex::one() % T::HistoryLength::get(),
-		);
-		NeighborRoots::<T, I>::insert((id, src_chain_id), neighbor_root_inx, root);
-		// Append new edge to the end of the edge list for the given tree
-		EdgeList::<T, I>::insert(id, src_chain_id, e_meta);
-		Ok(())
-	}
-
-	fn update_edge(
```
Comment (drewstone) [inline pallets/anchor/src/lib.rs:462]: Can we keep these functions mocked through the `T::LinkableTree` interface?

File: pallets/linkable-tree/src/types.rs [hunk 3]
```diff
@@ -0,0 +1,22 @@
+//! All the traits exposed to be used in other custom pallets
+use crate::*;
+use codec::{Decode, Encode};
+use scale_info::TypeInfo;
+
+#[derive(Default, Clone, Encode, Decode, TypeInfo)]
+pub struct AnchorMetadata<AccountId, Balance> {
```
Comment (drewstone) [inline pallets/linkable-tree/src/types.rs:7]: This probably isn't needed here? Metadata should be kept on `pallet-anchor`

**Review by drewstone — CHANGES_REQUESTED** [review]
Looks good so far!

**Comment by filiplazovic** [comment]
This is almost done, but I'm not sure we achieved our goal completely since edge updates are done through the anchor (the app). wdyt @drewstone

**Comment by drewstone** [comment]
I think I actually prefer it this way. I want the anchor to be where we
update add the edges but like the implementation is abstracted downstream.

**Comment by filiplazovic** [comment]
@drewstone This is ready

**Pull request merged**